sql/parser/builtins.go

```diff
@@ -66,9 +67,16 @@ func (b builtin) match(types typeList) bool {
 // sorted please.
 var builtins = map[string][]builtin{
 	// TODO(XisiHuang): support encoding, i.e., length(str, encoding).
-	"length": {stringBuiltin1(func(s string) (Datum, error) {
-		return DInt(len(s)), nil
-	}, DummyInt)},
+	"length": {
+		stringBuiltin1(func(s string) (Datum, error) {
+			return DInt(utf8.RuneCountInString(s)), nil
+		}, DummyInt),
+		bytesBuiltin1(func(s string) (Datum, error) {
+			return DInt(len(s)), nil
+		}, DummyInt),
+	},
+
+	// TODO(pmattis): What string functions should also support bytesType?
```

Comment:
I asked the question about supporting encodings other than utf8 in a PR a few weeks ago. Sounded like it wasn't important to do so. Would be a nice simplification to say that utf8 is our only supported encoding.